python/pyspark/sql/tests.py

```diff
@@ -1149,6 +1149,75 @@ def test_infer_schema(self):
         result = self.spark.sql("SELECT l[0].a from test2 where d['key'].d = '2'")
         self.assertEqual(1, result.head()[0])
 
+    def test_infer_schema_specification(self):
+        from decimal import Decimal
+
+        class A(object):
+            def __init__(self):
+                self.a = 1
+
+        data = [
+            True,
+            1,
+            "a",
+            u"a",
```

Comment:
nit: since this is for unicode string, how about using a non-ascii string?

Reply:
Fair point. Will change when I fix some codes around here.